### DISCUSSION TITLE
Installing php7.3-memcache uninstalls php-memcache

### DISCUSSION THREAD

**Discussion by Jemt** [oerdnj/deb.sury.org#2001] [Q&A]
Hi,

I recently realized that memcache was missing for PHP 7.3 on our Debian 10 server as it suddenly started to emit the following warning after an `apt-get upgrade`.

`PHP Warning:  PHP Startup: Unable to load dynamic library 'memcache.so'`

Memcache is installed for PHP 7.0, 7.4, 8.0, 8.1, and 8.2, so it's just missing for PHP 7.3, which happens to be the default version for Debian 10.

Until we upgrade to Debian 11, we'd like to stick with PHP 7.3. To eliminate the warning and fix the lack of support for memcache I'd like to install it. But when I do `apt-get install php7.3-memcache` it warns that php-memcache will be removed. This seems to be a meta package depending on php8.2-memcache.

Will I break anything - now or in the future - if I install `php7.3-memcache` and let apt-get remove the `php-memcache` package?

Thanks a lot in advance :-)

**Comment by oerdnj** [comment]
> Hi,
> 
> 
> 
> I recently realized that memcache was missing for PHP 7.3 on our Debian 10 server as it suddenly started to emit the following warning after an `apt-get upgrade`.
> 
> 
> 
> `PHP Warning:  PHP Startup: Unable to load dynamic library 'memcache.so'`
> 
> 
> 
> Memcache is installed for PHP 7.0, 7.4, 8.0, 8.1, and 8.2, so it's just missing for PHP 7.3, which happens to be the default version for Debian 10.
> 
> 
> 
> Until we upgrade to Debian 11, we'd like to stick with PHP 7.3. To eliminate the warning and fix the lack of support for memcache I'd like to install it. But when I do `apt-get install php7.3-memcache` it warns that php-memcache will be removed. This seems to be a meta package depending on php8.2-memcache.
> 
> 
> 
> Will I break anything - now or in the future - if I install `php7.3-memcache` and let apt-get remove the `php-memcache` package?
> 
> 
> 
> Thanks a lot in advance :-)

What is the output of:

apt-cache policy php7.3-memcache

?

**Comment by Jemt** [comment]
Hi @oerdnj.
Thank you for taking the time to respond back :-)
Here is the output:

```
root@isp:~# apt-cache policy php7.3-memcache
php7.3-memcache:
  Installed: (none)
  Candidate: 1:4.0.5.2-1+0~20230716.5+debian10~1.gbpe9883d
  Version table:
     1:4.0.5.2-1+0~20230716.5+debian10~1.gbpe9883d 500
        500 https://packages.sury.org/php buster/main amd64 Packages
```

**Reply by oerdnj** [reply]
> Hi @oerdnj.
> 
> Thank you for taking the time to respond back :-)
> 
> Here is the output:
> 
> 
> 
> ```
> 
> root@isp:~# apt-cache policy php7.3-memcache
> 
> php7.3-memcache:
> 
>   Installed: (none)
> 
>   Candidate: 1:4.0.5.2-1+0~20230716.5+debian10~1.gbpe9883d
> 
>   Version table:
> 
>      1:4.0.5.2-1+0~20230716.5+debian10~1.gbpe9883d 500
> 
>         500 https://packages.sury.org/php buster/main amd64 Packages
> 
> ```

And `php-memcache`?

**Comment by Jemt** [comment]
@oerdnj Here it is :-)

```
root@isp:~# apt-cache policy php-memcache
php-memcache:
  Installed: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
  Candidate: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
  Version table:
 *** 8.2-2+0~20230716.32+debian10~1.gbpbf55c7 500
        500 https://packages.sury.org/php buster/main amd64 Packages
        100 /var/lib/dpkg/status
     3.0.9~20170802.e702b5f-2 500
        500 http://deb.debian.org/debian buster/main amd64 Packages
```

**Reply by oerdnj** [reply]
> @oerdnj Here it is :-)
> 
> 
> 
> ```
> 
> root@isp:~# apt-cache policy php-memcache
> 
> php-memcache:
> 
>   Installed: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
> 
>   Candidate: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
> 
>   Version table:
> 
>  *** 8.2-2+0~20230716.32+debian10~1.gbpbf55c7 500
> 
>         500 https://packages.sury.org/php buster/main amd64 Packages
> 
>         100 /var/lib/dpkg/status
> 
>      3.0.9~20170802.e702b5f-2 500
> 
>         500 http://deb.debian.org/debian buster/main amd64 Packages
> 
> ```

That’s weird, I will have to take a closer look. Nevertheless, it’s ok to remove the dummy package.

**Reply by oerdnj** [reply]
> @oerdnj Here it is :-)
> 
> 
> 
> ```
> 
> root@isp:~# apt-cache policy php-memcache
> 
> php-memcache:
> 
>   Installed: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
> 
>   Candidate: 8.2-2+0~20230716.32+debian10~1.gbpbf55c7
> 
>   Version table:
> 
>  *** 8.2-2+0~20230716.32+debian10~1.gbpbf55c7 500
> 
>         500 https://packages.sury.org/php buster/main amd64 Packages
> 
>         100 /var/lib/dpkg/status
> 
>      3.0.9~20170802.e702b5f-2 500
> 
>         500 http://deb.debian.org/debian buster/main amd64 Packages
> 
> ```

That’s weird, I will have to take a closer look. Nevertheless, it’s ok to remove the dummy package.

**Comment by Jemt** [comment]
@oerdnj Cool, thanks a lot for your help :-)
